### PR TITLE
Fix tabulated pair potentials

### LINKED ISSUE #173: LAMMPS simulations failing when rmax is different than potential's cutoff
LAMMPS simulations are blowing up when the potential's stop value is larger than the rmax value. 

We should figure out what the largest rcut is for any pair, then use this as the cutoff for everyone & to reduce the number of points in the interpolation table

### PR DESCRIPTION
**Pull request by mphoward**
This PR should fix the way that potentials get their cutoff trimmed down. Before, the trimming was causing the tabulated potentials to be interpolated in LAMMPS when the cutoff didn't match the end of the table.

Here, we add a method to compute the pairwise energies and forces. Then, all of them are trimmed simultaneously to the biggest required cutoff.

Fixes #173

### CODE REVIEW

**Comment by clpetix** [comment]
This fixes the errors I was having in issue #173!

File: src/relentless/simulate/hoomd.py [hunk 1]
```diff
@@ -1006,8 +1005,10 @@ def process(self, sim, sim_op):
 
     def _get_rdf_params(self, sim):
         rdf_params = PairMatrix(sim.types)
-        rmax = sim.potentials.pair.x[-1]
+        rmax = sim[sim.initializer]["_potentials_rmax"]
```
Comment (mphoward) [inline src/relentless/simulate/hoomd.py:1008]: I think this change should be reverted, will think about it.

Comment (mphoward) [reply src/relentless/simulate/hoomd.py:1008]: I think this is OK for now.

**Comment by mphoward** [comment]
@clpetix this is ready for a final review

**Review by clpetix — APPROVED** [review]
This all looks good to me and has fixed the problem in my testing!

**Pull request merged**